src/main/kotlin/com/example/ajaxproject/service/GroupChatServiceImpl.kt

```diff
@@ -0,0 +1,123 @@
+package com.example.ajaxproject.service
+
+import com.example.ajaxproject.dto.request.CreateChatDto
+import com.example.ajaxproject.dto.request.GroupChatDTO
+import com.example.ajaxproject.dto.responce.GroupChatMessageResponse
+import com.example.ajaxproject.exeption.WrongActionException
+import com.example.ajaxproject.model.GroupChatMessage
+import com.example.ajaxproject.model.GroupChatRoom
+import com.example.ajaxproject.model.User
+import com.example.ajaxproject.repository.GroupChatMessageRepository
+import com.example.ajaxproject.repository.GroupChatRoomRepository
+import com.example.ajaxproject.service.interfaces.GroupChatService
+import com.example.ajaxproject.service.interfaces.UserService
+import com.example.ajaxproject.service.mapper.GroupChatMessageMapper
+import org.bson.types.ObjectId
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+class GroupChatServiceImpl(
+    private val groupChatRoomRepository: GroupChatRoomRepository,
+    private val userService: UserService,
+    private val groupChatMessageRepository: GroupChatMessageRepository,
+    private val groupChatMessageMapper: GroupChatMessageMapper
+) : GroupChatService {
+
+    companion object {
+        private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
+    }
+
+    override fun createGroupRoom(createChatDto: CreateChatDto): GroupChatRoom {
+
+        val user = userService.getUserById(createChatDto.adminId)
+
+        val groupChatRoom = GroupChatRoom(
+            id = ObjectId(),
+            chatName = createChatDto.chatName,
+            adminId = user.id,
+            chatMembers = listOf(user)
+        )
+
+        groupChatRoom.chatMembers + user
+
+        groupChatRoomRepository.save(groupChatRoom)
+
+        logger.info("Group chat room created successfully")
+
+        return groupChatRoom
+
+    }
+
+    override fun getAllChatMembers(chatId: String): List<User> {
+
+        val chat = groupChatRoomRepository.findChatRoom(chatId)
+
+        logger.info("Retrieved chat members for chat ID: {}" , chatId)
+
+        return chat.chatMembers
+    }
+
+    override fun addUserToChat(chatId: String, userId: String): List<User> {
+
+        val chat = groupChatRoomRepository.findChatRoom(chatId)
+
+        val user = userService.getUserById(userId)
+
+        chat.chatMembers += user
+
+        logger.info("User with ID {} added to chat ID {}" ,userId , chatId)
+
+        return groupChatRoomRepository.save(chat).chatMembers
+
+    }
```

Comment:
```suggestion
        return groupChatRoomRepository.save(chat).chatMembers
    }
```

fix everywhere. before closing brace should be no empty line